pkg/server/backuprestoreserver.go

```diff
@@ -328,6 +319,45 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 	return le.Run(ctx)
 }
 
+func (b *BackupRestoreServer) updatePeerURLIfChanged(ctx context.Context, tlsEnabled bool, logger *logrus.Logger) error {
+	logger.Info("Checking if peerURL has changed or not.")
+
+	m := member.NewMemberControl(b.config.EtcdConnectionConfig)
+
+	cli, err := etcdutil.NewFactory(*b.config.EtcdConnectionConfig).NewCluster()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := cli.Close(); err != nil {
+			b.logger.Errorf("failed to close etcd client: %v", err)
+		}
+	}()
+
+	changed, err := hasPeerURLChanged(ctx, m, cli)
+	if err != nil {
+		return err
+	}
+	if changed {
+		b.logger.Info("Etcd member peerURLs found to be changed.")
+		if err = retry.OnError(retry.DefaultBackoff, errors.IsErrNotNil, func() error {
+			if err = m.UpdateMemberPeerURL(ctx, cli); err != nil {
+				return err
+			}
+			b.logger.Info("Successfully updated the peerURLs for etcd member.")
+			return nil
+		}); err != nil {
+			return err
+		}
+		if err := miscellaneous.RestartEtcdWrapper(ctx, tlsEnabled, b.config.EtcdConnectionConfig); err != nil {
```

Comment:
do you wish to also retry this operation? If the restart does not succeed then some TCs will not pass as we have seen.

Reply:
TBH, calling a `stop` endpoint within the same network shouldn't fail in the first attempt itself as it's not a api call, so if it somehow fails then I don't think retry will help as we already exiting the container and restarting it.
Moreover, I also don't want to add a retry everywhere, I want to a retry backoff mechanism where ever necessary.